core/src/components/datetime/utils/data.ts

```diff
@@ -166,17 +167,24 @@ export const generateTime = (
       if (minParts.hour !== undefined) {
         processedHours = processedHours.filter(hour => {
           const convertedHour = refParts.ampm === 'pm' ? (hour + 12) % 24 : hour;
-          return convertedHour >= minParts.hour!;
+          return (use24Hour ? hour : convertedHour) >= minParts.hour!;
         });
         isAMAllowed = minParts.hour < 13;
       }
       if (minParts.minute !== undefined) {
-        processedMinutes = processedMinutes.filter(minute => minute >= minParts.minute!);
+        processedMinutes = processedMinutes.filter(minute => {
+          if (minParts.hour !== undefined && refParts.hour !== undefined) {
+            if (refParts.hour > minParts.hour) {
+              return true;
+            }
+          }
```

Comment:
Could we pull these conditions out into a separate variable outside the loop? Would improve readability and performance.

Reply:
Great call 👍